Documentation/general (#103)
Kryha/KREAd: docs/glossary.md

#### Semi-fungible token (SFT)

> Behaves like a non-fungible token, meaning it's a digital asset that can hold arbitrary data. However, unlike NFTs multiple SFTs with the same data can be minted. For example, 10 SFTs with data `{ type: "background" }` can exist, while in the case of NFTs only a single NFT can exist with the exact same data. The KREAd contract can mint two SFTs brands: [KREAd Characters](#character-sft) and KREAd Items.

Reviewer: ```suggestion
Behaves like a non-fungible token, meaning it's a digital asset that can hold arbitrary data. However, unlike NFTs multiple SFTs with the same data can be minted. For example, 10 SFTs with data `{ type: "background" }` can exist, while in the case of NFTs only a single NFT can exist with the exact same data. The KREAd contract can mint two SFT brands: [KREAd Characters](#character-sft) and KREAd Items.
```

link to outside source for brands?

| artistMetadata  | Link to the artists Instagram account | string (link to IG page)  | 'https://www.instagram.com/enmanueljrperez/'                                                                        |
| image           | Link to the character image           | string (link to webp file) | 'https://pink-defensive-jay-557.mypinata.cloud/ipfs/Qmc55m7RrzZtB25mM9B2c9BdtXcLDbhSrErWMAa7azCet6/Citizen/images/001.webp' |
| characterTraits | Link to the character traits          | string (link to json file) | 'https://pink-defensive-jay-557.mypinata.cloud/ipfs/Qmc55m7RrzZtB25mM9B2c9BdtXcLDbhSrErWMAa7azCet6/Citizen/metadata.json'   |

```js
CharacterSFT = {
  "artistMetadata": "https://instagram.com/enmanueljrperez?igshid=MzRlODBiNWFlZA==",
  "characterTraits": "",
  "date": {
    "absValue": "1700056310n",
    "timerBrand": {}
  },
  "description": "A Citizen from the Wildlands.",
  "id": 95,
  "image": "QmR16ndHynQmrRLXMFV5YkJGUYA1BKGij2PKU9JsnJmMTy/08.png",
  "keyId": 1,
  "level": 1,
  "name": "juanito",
  "origin": "Wildlands",
  "title": "Citizen",
  "url": "https://builder.agoric.kryha.dev/static/media/default-character.216ad02c.png"
}
```

> > 💡 Properties containing links do not include the full url, prefix them with `https://pink-defensive-jay-557.mypinata.cloud/ipfs/` if you wish to access the link

Reviewer: ```suggestion
> 💡 Properties containing IPFS cids do not include the full url, prefix them with `https://pink-defensive-jay-557.mypinata.cloud/ipfs/` if you wish to access the document
```

| image          | Link to the character image                                                                                           | string (link to webp file) | 'pink-defensive-jay-557.mypinata.cloud/ipfs/Qmc55m7RrzZtB25mM9B2c9BdtXcLDbhSrErWMAa7azCet6/Citizen/images/001.webp' |

```js
ItemSFT = {
  "artistMetadata": "https://instagram.com/enmanueljrperez?igshid=MzRlODBiNWFlZA==",
  "category": "background",
  "description": "Like an unspoken language, it communicates emotion, directs attention, and influences perception. Whether vibrant, muted, or in between, it's the silent foundation of visual harmony.",
  "image": "https://pink-defensive-jay-557.mypinata.cloud/ipfs/QmfW1CGtyn3t5MUfwk86n936og7Q3ruhKda5nu6ExJAp8N/Sky%20gray.png",
  "name": "Sky gray",
  "origin": "Mount",
  "rarity": 18,
  "thumbnail": "https://pink-defensive-jay-557.mypinata.cloud/ipfs/QmfW1CGtyn3t5MUfwk86n936og7Q3ruhKda5nu6ExJAp8N/Sky%20gray.png",
}
```

> > 💡 Properties containing links do not include the full url, prefix them with `https://pink-defensive-jay-557.mypinata.cloud/ipfs/` if you wish to access the link

Reviewer: ```suggestion
> 💡 Properties containing IPFS cids do not include the full url, prefix them with `https://pink-defensive-jay-557.mypinata.cloud/ipfs/` if you wish to access the document
```

An equip offer looks like this:

```js
const equipProposal = {
    // the following assets are sent from your wallet to Zoe
    give: {
        // the character SFT you wish to equip to, it's temporarily sent to Zoe as proof of ownership and will be returned
        CharacterKey2: {
            brand: characterBrand,
            value: makeCopyBag([[characterToEquipTo, 1n]]),
        },
        // the item SFT you wish to equip
        Item: {
            brand: itemBrand,
            value: makeCopyBag([[itemToEquip, 1n]]),
        },
    },
    // the following assets must be received or the give will be returned
    want: {
        // the character must be returned
        CharacterKey1: { brand: characterBrand, value: makeCopyBag([[characterToEquipTo, 1n]]) },
    }
};

```

#### Unequip

> The unequip action allows the owner of a [Character](#character-sft) to transfer an item from the Character's Inventory to their own wallet. Only one Item can be unequipped at a time and the only requiremnt is ownership of the Character. A user may want to unequip an item for the following reasons:

Reviewer: ```suggestion
The unequip action allows the owner of a [Character](#character-sft) to transfer an item from the Character's Inventory to their own wallet. Only one Item can be unequipped at a time and the only requirement is ownership of the Character. A user may want to unequip an item for the following reasons:
```

   
A swap proposal looks like this:
```js
const swapProposal = {
    // the following assets are sent from your wallet to Zoe
    give: {
        // the character SFT you wish to perform a swap on, it's temporarily sent to Zoe as proof of ownership and will be returned
        CharacterKey1: { brand: charBrand, value: makeCopyBag([[characterToEquipTo, 1n]]) },
         // the item SFT you wish to equip (must be the same category as proposal.want.Item)
        Item: {
            brand: itemBrand,
            value: makeCopyBag([[wantedItem, 1n]]),
        },
    },
    // the following assets must be received or the give will be returned
    want: {
        // the character must be returned
        CharacterKey2: {
            brand: charBrand,
            value: makeCopyBag([[characterToEquipTo, 1n]]),
        },
        // the item SFT you wish to unequip
        Item: {
            brand: itemBrand,
            value: makeCopyBag([[wantedItem, 1n]]),
        },
    },
};
```

> #### Item Category

Reviewer: maybe move this part more up?

Author: moved directly below inventory

> Item SFTs have a property `category` which determines how the item will fit (visually) when equipped onto a character's inventory. For example, items of category "mask" will be rendered on top of the character in the position of the character's mouth, while "background" items will be in the center behind the character. Character's are not required to have any items equipped and can only have a single item of a given category at once. Note that all KREAd items share a brand and issuer, despite the added functionality based on its property `category`. Currently, KREAd Items must be one of the following categories:

Reviewer: ```suggestion
Item SFTs have a property `category` which determines how the item will fit (visually) when equipped onto a character's inventory. For example, items of category `mask` will be rendered on top of the character in the position of the character's mouth, while `background` items will be in the center behind the character. Character's are not required to have any items equipped and can only have a single item of a given category at once. Note that all KREAd items share a brand and issuer, despite the added functionality based on its property `category`. Currently, KREAd Items must be one of the following categories:
```

#### Semi-fungible token (SFT)
Behaves like a non-fungible token, meaning it's a digital asset that can hold arbitrary data. However, unlike NFTs multiple SFTs with the same data can be minted. For example, 10 SFTs with data `{ type: "background" }` can exist, while in the case of NFTs only a single NFT can exist with the exact same data. The KREAd contract can mint two SFT [brands](https://docs.agoric.com/glossary/#brand): [KREAd Characters](#character-sft) and [KREAd Items](#item-sft).

#### Character SFT

> [Semi-fungible token](#semi-fungible-token-sft) representing a character in the SAGES universe. Each character can be identified by a unique image and a set of properties related to the SAGES story. Characters can be minted for a fee via KREAd's frontend. When minting, the user must choose a valid name for the character, and will reviece a randomly selected character from a predefined set. On their own, Character assets behave like any other SFT on Agoric, meaning they can be used in offers and be recognized by other contracts. For example they can be sold and bought by anyone using the marketplace section of KREAd's frontend, or they can be sent to a different marketplace contract on the Agoric chain that handles SFTs. KREAd extends the functionality of Characters by providing an Inventory in which to store [KREAd Items](#item-sft). 

Reviewer 2: ```suggestion
[Semi-fungible token](#semi-fungible-token-sft) representing a character in the SAGES universe. Each character can be identified by a unique image and a set of properties related to the SAGES story. Characters can be minted for a fee via KREAd's frontend. When minting, the user must choose a valid name for the character, and will receive a randomly selected character from a predefined set. On their own, Character assets behave like any other SFT on Agoric, meaning they can be used in offers and be recognized by other contracts. For example they can be sold and bought by anyone using the marketplace section of KREAd's frontend, or they can be sent to a different marketplace contract on the Agoric chain that handles SFTs. KREAd extends the functionality of Characters by providing an Inventory in which to store [KREAd Items](#item-sft). 
```

| weight         | TODO                                                                                                                  | number                     | 0                                                                                                                   |
| colors         | List of colors present in the item, used for filters                                                                  | string []                  | []                                                                                                                  |
| reserves       | TODO                                                                                                                  | number                     | 0                                                                                                                   |
| functional     | TODO                                                                                                                  | boolean                    | false                                                                                                               |

> | artistMetadata | Link to the artists Instagram account                                                                                 | string (linkt to IG page)  | 'https://www.instagram.com/enmanueljrperez/'                                                                        |

Reviewer 2: ```suggestion
| artistMetadata | Link to the artists Instagram account                                                                                 | string (link to IG page)  | 'https://www.instagram.com/enmanueljrperez/'                                                                        |
```

1. To change the properties of a Character (look and stats)
2. To manually equip a different item of the same category (consider using [swap](#swap) in this situation)
3. To sell the item in the marketplace
   
An unequip proposal looks like this:
```js
const unequipProposal = {
    // the following assets are sent from your wallet to Zoe
    give: {
        // the character SFT you wish to equip to, it's temporarily sent to Zoe as proof of ownership and will be returned
        CharacterKey1: { brand: charBrand, value: makeCopyBag([[characterToEquipTo, 1n]]) },
    }
    // the following assets must be received or the give will be returned
    want: {
        // the character must be returned
        CharacterKey2: {
            brand: charBrand,
            value: makeCopyBag([[characterToEquipTo, 1n]]),
        },
        // the item SFT you wish to unequip
        Item: {
            brand: itemBrand,
            value: makeCopyBag([[wantedItem, 1n]]),
        },
    },
};
```

#### Swap

> The swap action allows the owner of a [Character](#character-sft) to swap an [Inventory](#inventory) item for one from their wallet. Since only one Item per [category](#item-category) can be equipped at a time, replacing an equipped item would require two separate transaction (first unequip, then equip new item). The swap action combines it into a single transaction and allows users to replace an equipped item for anotherone of the same category. 

Reviewer 2: ```suggestion
The swap action allows the owner of a [Character](#character-sft) to swap an [Inventory](#inventory) item for one from their wallet. Since only one Item per [category](#item-category) can be equipped at a time, replacing an equipped item would require two separate transaction (first unequip, then equip new item). The swap action combines it into a single transaction and allows users to replace an equipped item for another one of the same category. 
```

```js
// for makeSellItemInvitation
const proposal {
    want : {
        Price: { brand: IST_BRAND, value: ASKING_PRICE_IN_IST },
    },
    give: {
        Item: { brand: KREAD_ITEM_BRAND, value: ITEM_TO_SELL },
    }
}

// for makeSellCharacterInvitation
const proposal {
    want : {
        Price: { brand: IST_BRAND, value: ASKING_PRICE_IN_IST },
    },
    give: {
        Item: { brand: KREAD_CHARACTER_BRAND, value: CHARACTER_TO_SELL },
    }
}
```
>Assets specified on an Offer's give will be escrowed by Zoe (Agoric's smart contract platform) once they sign the offer transaction, meaning they will no longer be in the user's wallet the unless they cancel the offer before completion.

Once the market listing is created, it will be visible at kread.app/shop, where any user can buy it (provided they have enough IST to cover the set asking price + fees). The user sets the asking price via the offer's want property, then Agoric's [Offer Safety](https://docs.agoric.com/glossary/#offer-safety) guarantees they will either receive the asking price and transfer the asset to the buyer, or maintain the right to cancel the offer (and receive their asset back). 

> KREAd does not impose a time limit to market entries, as long as its creator does not cancel it it will be available for a counter party to fulfill the offer.

Reviewer 2: ```suggestion
KREAd does not impose a time limit to market entries. As long as its creator does not cancel it, it will be available for a counter party to fulfill the offer.
```
